private/buf/bufformat/formatter.go

```diff
@@ -1009,6 +1039,10 @@ func (f *formatter) writeRange(rangeNode *ast.RangeNode) {
 //	  json_name = "something"
 //	]
 func (f *formatter) writeCompactOptions(compactOptionsNode *ast.CompactOptionsNode) {
+	f.inCompactOptions = true
+	defer func() {
+		f.inCompactOptions = false
+	}()
```

Comment:
This is kind of gross, to stash this state on `f` instead of on the stack. But weaving this flag all the way through numerous function calls, to get it into `writeOptionName` was even messier. So I went with the less smelly approach.